bin/get-video-s3-acls.py

```diff
@@ -0,0 +1,302 @@
+
+
+# Get all keys from AWS S3
+def get_s3_bucket_raw_keys_list(s3_bucket=AWS_S3_BUCKET):
+    print(f"Getting raw AWS S3 keys recursively ({s3_bucket}) ...", file=sys.stderr)
+    result = subprocess.run(
+        [
+            AWSCLI,
+            "s3",
+            "ls",
+            f"s3://{s3_bucket}",
+            "--recursive",
+        ],
+        env=os.environ,
+        capture_output=True,
+        check=True,
+        text=True,
+    )
+
+    # Separate out just the key from date, time, size, key
+    this_s3_bucket_raw_keys_list = []
+    for line in result.stdout.split("\n"):
+        if line:
+            this_s3_bucket_raw_keys_list.append(re.split(r"\s+", line, 3)[3])
+
+    print(
+        f"{len(this_s3_bucket_raw_keys_list)} rows retrieved",
+        file=sys.stderr,
+    )
+
+    return this_s3_bucket_raw_keys_list
+
+
+# Get the video files info from NZSL Signbank
+def get_nzsl_raw_keys_dict():
+    this_nzsl_raw_keys_dict = {}
+    print(
+        f"Getting raw list of video file info from NZSL Signbank ...",
+        file=sys.stderr,
+    )
+    result = subprocess.run(
+        [
+            PGCLI,
+            "-c",
+            "COPY (SELECT id AS db_id, gloss_id, is_public, videofile FROM video_glossvideo) "
+            "TO STDOUT WITH (FORMAT CSV)",
+            f"{DATABASE_URL}",
+        ],
+        env=os.environ,
+        capture_output=True,
+        check=True,
+        text=True,
+    )
+
+    # Separate out the NZSL db columns
+    # Write them to a dictionary, so we can do fast operations
+    for rawl in result.stdout.split("\n"):
+        rawl = rawl.strip()
+        if not rawl:
+            continue
+        [db_id, gloss_id, is_public, video_key] = rawl.split(",")
+        this_nzsl_raw_keys_dict[video_key] = [db_id, gloss_id, is_public.lower() == "t"]
+
+    print(
+        f"{len(this_nzsl_raw_keys_dict)} rows retrieved",
+        file=sys.stderr,
+    )
+
+    return this_nzsl_raw_keys_dict
+
+
+# Get the s3 keys present and absent from our NZSL keys
+def create_all_keys_dict(this_s3_bucket_raw_keys_list, this_nzsl_raw_keys_dict):
+    print("Getting S3 keys present and absent from NZSL Signbank ...", file=sys.stderr)
+    nkeys_present = 0
+    nkeys_absent = 0
+    this_all_keys_dict = {}
+    for video_key in this_s3_bucket_raw_keys_list:
+        if video_key in this_nzsl_raw_keys_dict:
+            nkeys_present += 1
+            # Add 'Present' column to start
+            this_all_keys_dict[video_key] = [True] + this_nzsl_raw_keys_dict[video_key]
+        else:
+            nkeys_absent += 1
+            # Add 'Present' (absent) column to start
+            this_all_keys_dict[video_key] = [False, "", "", ""]
+
+    print(f"PRESENT: {nkeys_present} keys", file=sys.stderr)
+    print(f"ABSENT:  {nkeys_absent} keys", file=sys.stderr)
+
+    # Write all keys back to a cache file
+    with open(ALL_KEYS_CACHE_FILE, "w") as f_obj:
+        for video_key, item_list in this_all_keys_dict.items():
```

Comment:
suggestion: it probably won't be measurable, but you might as well merge this loop with the previous one to save a bit of looping

Reply:
Done.